pages/api/cart.js

```diff
@@ -0,0 +1,113 @@
+import { serialize, parse } from 'cookie'
+import isAllowedMethod from '../../lib/api/is-allowed-method'
+import DB from '../../lib/api/db'
+
+const METHODS = ['GET', 'POST']
+const SESSION = 'session-id'
+const ONE_DAY = 60 * 60 * 24
+const MAX_AGE = ONE_DAY * 90
+
+async function getSessionId(req, res) {
+  let sessionId = req.cookies[SESSION]
+
+  if (!sessionId) {
+    const session = await DB.createSession()
+
+    sessionId = session.id
+
+    const cookie = serialize(SESSION, sessionId, {
+      maxAge: MAX_AGE,
+      expires: new Date(Date.now() + MAX_AGE * 1000),
+      httpOnly: true,
+      secure: process.env.NODE_ENV === 'production',
+      path: '/',
+      sameSite: 'lax',
+    })
```

Comment:
This cookie will only be available for the server, the client will only be able to access the cart data by making a `GET` request to this endpoint, that translates in more security and more control for us.

The cookie will last for `3 months` but can be revalidated so it never expires.

The session id is saved directly, it's also possible to use a salt so it can't be changed by the user.